lib/utils/fields.js

```diff
@@ -206,15 +214,28 @@ export function slotTitleForNameMap(fields) {
 export function findFieldIndex(fields, key, translationMap = {}) {
   // First try to find a direct match.
   let index = fields.findIndex((f) => {
-    return f.name === key || f.alias === key;
+    // Consider the following situation:
+    // `key === undefined`
+    // `f = {'name': 'a', 'title': 'a'}`
+    // If we do not check whether `f` has `name` or `alias`, then:
+    // 1) `f.name === key || f.alias === key`
+    // 2) `'a' === undefined || undefined === undefined`
+    // 3) `false || true`
+    // 4) `true` --> direct match
+    // i.e., we directly match `undefined` to a defined field with name `a`
```

Comment:
This was not directly causing a failing test, but it is an issue I identified while debugging some other code that was causing a failing test. There should have been a console warning for that other failing test "Failed to direct match ...", but it was not being emitted for the failing test due to the above bug.

Reply:
We were erroneously saying an undefined key was a direct match to some defined fields that were simply missing an alias attribute.